cpp/dolfinx/common/MPI.h

```diff
@@ -74,6 +74,13 @@ int rank(MPI_Comm comm);
 /// communicator
 int size(MPI_Comm comm);
 
+/// @brief Checks wether an error code returned by an MPI
+/// function is equal to MPI_SUCCESS. If the check fails then
+/// throw a runtime error.
```

Comment:
'runtime error' is a bit vague. Does it just 'abort'?

Reply:
Yep, I need to review documentation and function name.
I was throwing a runtime error before, but it can cause a deadlock when `MPI_Abort` fails.
This can happen when the communicator is NULL (
reusing a communicator that has been freed) followed by a barrier with `MPI_COMM_WORLD`.
So I think a sensible solution is to forcibly abort the execution while still printing the error message.